src/CacheManager.Core/Internal/CacheKeysHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace CacheManager.Core.Internal
+{
+    /// <summary>
+    /// Helper functions for manipulating keys.
+    /// </summary>
+    static public class CacheKeysHelper
```

Comment:
Ah ok, does that go away if you move the main implementation into BaseCacheHandle?
Otherwise, let's try to build it without an extension method, then, a public helper is fine. But extension methods also extend into consumer code...